File: LeetCode/56. Merge Intervals.py
### KNOWLEDGE IS POWER  ###

class Solution(object):
    def merge(self, intervals):
        """
        :type intervals: List[List[int]]
        :rtype: List[List[int]]
        """
        intervals.sort()
        ans = []
        for [x, y] in intervals:
            if ans:
                if x <= ans[-1][1] < y:
                    ans[-1][1] = y
                elif x > ans[-1][1]:
                    ans.append([x, y])
            else:
                ans.append([x, y])
        return ans

File: LeetCode/56. Merge Intervals.cpp
/***  KNOWLEDGE IS POWER  ***/

class Solution {
public:
    vector<vector<int>> merge(vector<vector<int>>& intervals) {
        sort(intervals.begin(), intervals.end()) ;
        vector<vector<int>> ans ;
        for (int i = 0; i < intervals.size(); i ++) 
        {
            int x = intervals[i][0], y = intervals[i][1];
            if(ans.size()) 
            {
                if (x <= ans.back()[1] && ans.back()[1] < y)
                    ans.back()[1] = y;
                else if (intervals[i][0] > ans.back()[1])
                    ans.push_back(intervals[i]);
            } 
            else 
            {
                ans.push_back(intervals[i]);
            }
        }
        return ans;
    }
};
